docs/source/dataframe-hive.rst

```diff
@@ -0,0 +1,227 @@
+.. _dataframe.hive:
+
+Using Hive Partitioning with Dask
+=================================
+
+.. currentmodule:: dask.dataframe
+
+It is sometimes useful to write your dataset with a hive-like directory scheme.
+For example, if your dataframe contains ``'year'`` and ``'semester'`` columns,
+a hive-partitioned directory scheme might look something like the following::
+
+    output-path/
+    ├── year=2022/
+    │   ├── semester=fall/
+    │   │   └── part.0.parquet
+    │   └── semester=spring/
+    │       ├── part.0.parquet
+    │       └── part.1.parquet
+    └── year=2023/
+        └── semester=fall/
+            └── part.1.parquet
+
+The use of this self-describing scheme implies that all rows within the
+``'output-path/year=2022/semester=fall/'`` directory will contain the
+value ``2022`` in the ``'year'`` column and the value ``'fall'`` in the
+``'semester'`` column. This means that the parquet files themselves can
+(optionally) exclude the ``'year'`` and ``'semester'`` columns entirely.
+
+The primary advantage of generating a hive-partitioned dataset
+is that certain IO filters can be applied by :func:`read_parquet`
+without the need to parse any footer metadata. In other words,
+the following command will typically be faster when the dataset
+is already hive-partitioned on the ``'year'`` column.
+
+.. code-block:: python
+
+    >>> dd.read_parquet("output-path", filters=[("year", ">", 2022)])
+
+Writing Parquet Data with Hive Partitioning
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Dask's :func:`to_parquet` function will produce a hive-partitioned
+directory scheme automatically when the ``partition_on`` option is used.
+
+.. code-block:: python
+
+    >>> df.to_parquet("output-path", partition_on=["year", "semester"])
+
+    >>> os.listdir("output-path")
+    ["year=2022", "year=2023"]
+
+    >>> os.listdir("output-path/year=2022")
+    ["semester=fall", "semester=spring"]
+
+    >>> os.listdir("output-path/year=2022/semester=spring")
+    ['part.0.parquet', 'part.1.parquet']
+
+
+It is important to recognize that Dask will **not** aggregate the
+data files written within each of the leaf directories. This is
+because each of the DataFrame partitions is written independently 
+during the execution of the :func:`to_parquet` task graph. In order 
+to write out data for partition `i`, the partition-i write task will
+perform a `groupby` operation on columns ``["year", "semester"]``,
+and then each distinct group will be written to the corresponding
+directory using the file name ``'part.{i}.parquet'``. Therefore, it
+is possible for a hive-partitioned write to produce a large number
+of files in every leaf directory (one for each DataFrame partition).
```

Comment:
I've encountered situations where it was desired to have few (if not exactly one) files. This is doable with

```python
partition_on = ["year", "semester"]
df.shuffle(on=partition_on).to_parquet(partition_on=partition_on)
```

which is obviously expensive but is easy to do if a user truly cares about this. Is this a fringe use case or worth mentioning?

Reply:
This actually seems to come up a lot in practice, so I think it would be okay to add a note.